modify /etc/os-release

jpculp opened the pull request:
**Issue number:**

N/A

**Description of changes:**

Adds **HOME_URL**, **SUPPORT_URL**, and **BUG_REPORT_URL** to `/etc/os-release`,
as well as the _variant_ to the **PRETTY_NAME** and a new **VERSION** field.

**OLD**
```
NAME=Bottlerocket
ID=bottlerocket
PRETTY_NAME="Bottlerocket OS 1.2.0"
VARIANT_ID=aws-ecs-1
VERSION_ID=1.2.0
BUILD_ID=ccf1b754
```

**NEW**
```
NAME=Bottlerocket
ID=bottlerocket
VERSION="1.2.0 (aws-ecs-1)"
PRETTY_NAME="Bottlerocket OS 1.2.0 (aws-ecs-1)"
VARIANT_ID=aws-ecs-1
VERSION_ID=1.2.0
BUILD_ID=e0808b0b
HOME_URL="https://github.com/bottlerocket-os/bottlerocket"
SUPPORT_URL="https://github.com/bottlerocket-os/bottlerocket/discussions"
BUG_REPORT_URL="https://github.com/bottlerocket-os/bottlerocket/issues"
```

**Testing done:**

Built and launched `aws-k8s-1.20` and `aws-ecs-1` variants. Verified the expected changes via the admin container. Both instances connected to their respective clusters without issue, although the **OS Image** now reports as `Bottlerocket OS 1.2.0 (aws-k8s-1.20)` in EKS.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

tjkirch commented on tools/rpm2img:
```diff
@@ -152,10 +152,15 @@ mv "${ROOT_MOUNT}/boot"/* "${BOOT_MOUNT}"
 SYS_ROOT="${ARCH}-bottlerocket-linux-gnu/sys-root"
-echo "PRETTY_NAME=\"${PRETTY_NAME} ${VERSION_ID}\"" >> "${ROOT_MOUNT}/${SYS_ROOT}/usr/lib/os-release"
+VERSION="${VERSION_ID} (${VARIANT})"
+echo "VERSION=\"${VERSION}\"" >> "${ROOT_MOUNT}/${SYS_ROOT}/usr/lib/os-release"
```
nit: probably time for a `cat <<EOF >>${ROOT_MOUNT}/${SYS_ROOT}/usr/lib/os-release` rather than repeating echo/filename every time.

jpculp committed:
Add HOME_URL, SUPPORT_URL, and BUG_REPORT_URL to /etc/os-release

jpculp committed:
Add variant to PRETTY_NAME in /etc/os-release

This commit creates a VERSION field which is comprised of the VERSION_ID
and the variant in parenthesis. The VERSION field replaces VERSION_ID in
PRETTY_NAME.

jpculp commented:
Replaced the multiple echos with a single `cat <<EOF >>`, which is much easier to read.

zmrow reviewed (approved):
:lollipop:

bcressey commented on tools/rpm2img:
```diff
@@ -152,10 +152,17 @@ mv "${ROOT_MOUNT}/boot"/* "${BOOT_MOUNT}"
-echo "BUILD_ID=${BUILD_ID}" >> "${ROOT_MOUNT}/${SYS_ROOT}/usr/lib/os-release"
+VERSION="${VERSION_ID} (${VARIANT})"
+cat <<EOF >> "${ROOT_MOUNT}/${SYS_ROOT}/usr/lib/os-release"
+VERSION="${VERSION}"
+PRETTY_NAME="${PRETTY_NAME} ${VERSION}"
+VARIANT_ID=${VARIANT}
+VERSION_ID=${VERSION_ID}
```
nit: should we also add `VARIANT`? Might be good to keep related fields together too.
```suggestion
PRETTY_NAME="${PRETTY_NAME} ${VERSION}"
VARIANT="${VARIANT}"
VARIANT_ID="${VARIANT}"
VERSION="${VERSION}"
VERSION_ID=${VERSION_ID}
```

tjkirch replied:
I'd hesitate to add VARIANT because it blurs the lines between a field and its corresponding _ID field, where _ID is the "technical" name... if we come up with "friendly" versions of variant names, I think that's what would go into VARIANT, but we don't have those now...

jpculp replied:
I've typically seen VERSION close to the top of the os-release file (perhaps to make it less buried?)

Regarding VARIANT, I think I agree with @tjkirch that it would make more sense if it were a friendly name. That said, there is no reason we can't draft some up while we're at it.

bcressey replied:
It feels like we use our variant identifiers interchangeably with names, and I'm not sure how they could be made more friendly.

I don't feel strongly about it, and I'm willing to hold off on `VARIANT` to leave the door open.
